mobile/screens/settings/EditAccount.tsx

```diff
@@ -0,0 +1,351 @@
+          if (auth.currentUser && auth.currentUser.email) {
+            setEmail(auth.currentUser.email);
+          }
+        });
+    }
+    updateDoc(caregiverDoc, {
+      firstName: first,
+      lastName: last,
+      phoneNumber: phone,
+      email: email,
+    });
+  }
+
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity
+        style={{ paddingLeft: 10, paddingTop: 10 }}
+        onPress={() => {
+          setModalVisible(true);
+        }}
+      >
+        <AntDesign name="left" size={30} color="black" />
+      </TouchableOpacity>
+      <ScrollView showsVerticalScrollIndicator={false}>
+        <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
+          <View style={styles.inner}>
+            <Text style={styles.title}>Edit Account Information</Text>
+            <Text style={styles.description}>First Name</Text>
+            <TextInput
+              autoFocus={true}
+              style={[styles.input, firstEmpty && { borderColor: "#FF3939" }]}
+              onChangeText={(first) => {
+                first !== "" && setFirstEmpty(false);
+                setFirst(first);
+              }}
+              defaultValue={authData?.caregiver?.firstName}
+            />
+            {firstEmpty && <RequiredField />}
+            <Text style={styles.description}>Last Name</Text>
+            <TextInput
+              autoFocus={true}
+              style={[styles.input, lastEmpty && { borderColor: "#FF3939" }]}
+              onChangeText={(last) => {
+                last !== "" && setLastEmpty(false);
+                setLast(last);
+              }}
+              defaultValue={authData?.caregiver?.lastName}
+            />
+            {lastEmpty && <RequiredField />}
+            <Text style={styles.description}>Email</Text>
+            <TextInput
+              placeholder="email"
+              autoCompleteType="email"
+              keyboardType="email-address"
+              autoCapitalize="none"
+              autoFocus={true}
+              style={[styles.input, emailEmpty && { borderColor: "#FF3939" }]}
+              onChangeText={(email) => {
+                email !== "" && setEmailEmpty(false) && setPasswordEmpty(false);
+                setEmail(email);
+              }}
+              defaultValue={authData?.caregiver?.email}
+            />
+            {emailEmpty && <RequiredField />}
+            {email != (authData?.caregiver?.email as string) && (
+              <View>
+                <Text style={[styles.description, { fontSize: 14 }]}>
+                  Please confirm your password
+                </Text>
+                <TextInput
+                  style={[
+                    styles.input,
+                    passwordEmpty && { borderColor: "#FF3939" },
+                  ]}
+                  onChangeText={(password) => {
+                    password !== "" && setPasswordEmpty(false);
+                    setPassword(password);
+                  }}
```

Comment:
A quicker way to do this (though I might be missing some things):

```tsx
style={[
  styles.input,
  password.length === 0 && {borderColor: "#FF3939"}
]{
```

That way you don't (AFAIK) need `passwordEmpty` and `setPasswordEmpty`!

Reply:
fixed!